students/hyska_monika/lesson_10_organizing_files/deleting_unneeded_files.py

```diff
@@ -0,0 +1,45 @@
+"""
+Program  walks through a folder tree and searches for
+exceptionally large files or folders, eg. size > 8 mb
+Program delete found files and folders
+"""
+import shutil
+import os
+
+
+# function search folders and files bigger then some size
+def search_files_folders_bigger_size(path, size):
+    size_in_bytes = size * 1048576
+    files_list = []
+    folders_list = []
+    for folder_name, sub_folders, file_names in os.walk(path):
+        folder_size = 0
+        for file_name in file_names:
+            file_name_with_path = os.path.join(folder_name, file_name)
+            if os.path.getsize(file_name_with_path) > size_in_bytes:
+                files_list.append(os.path.abspath(file_name_with_path))
+            folder_size += os.path.getsize(file_name_with_path)
+        if folder_size > size_in_bytes:
+            folders_list.append(os.path.abspath(folder_name))
```

Comment:
The content of the for has grown large enough that you could try to subdivide it into a function. PyCharm has great refactoring tools, you just select a piece of code, choose refactor > extract and it suggests parameters and output. The value from subdividing the code is that you can see which of the scopes variables are needed. For example you are not using `sub_folders` at all (you could rename it to `__`, by the way `_` is for gettext in translation tools by convention).

Reply:
I assume you are right that they meant folders having more than 100MB files in them _directly_ and that we don't sum over subfolders.